### PR TITLE
O3-3310: Adding a modal for deleting a packaging unit

### PR DESCRIPTION
**Pull request by Daphne210**
https://github.com/openmrs/openmrs-esm-stock-management/assets/89844768/a6e5b232-541a-42f7-bbb0-75965c14c485

https://openmrs.atlassian.net/browse/O3-3310

### CODE REVIEW

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx [hunk 1]
```diff
@@ -119,54 +121,56 @@ const PackagingUnits: React.FC<PackagingUnitsProps> = ({
...
-                      {...getHeaderProps({
-                        header,
-                        isSortable: false,
-                      })}
-                      style={header.styles}
-                      key={header.key}
-                    >
-                      {header.header?.content ?? header.header}
-                    </TableHeader>
+    <>
```
Comment (jabahum) [inline src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx:124]: do we really need this tag

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx [hunk 2]
```diff
@@ -179,6 +183,23 @@ const PackagingUnitRow: React.FC<{
 }> = ({ isEditing, row, key }) => {
   const { t } = useTranslation();
```
Comment (jabahum) [inline src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx:185]: lets stick o3 implementation style of modals and dialogs

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx [hunk 3]
```diff
@@ -213,43 +234,73 @@ const PackagingUnitRow: React.FC<{
...
+              className="submitButton clear-padding-margin"
+              iconDescription={"Delete"}
+              kind="ghost"
+              renderIcon={TrashCan}
+              onClick={() => launchDeleteModal(row.packagingUomName)}
+            />
+          </div>
+        </TableCell>
+      </TableRow>
+      {showModal && (
```
Comment (jabahum) [inline src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx:276]: stick to o3 implementation of dialog and modals

**Review by jabahum — CHANGES_REQUESTED** [review]

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx (hunk 3 above)
Comment (Daphne210) [reply src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx]: > stick to o3 implementation of dialog and modals

This has been addressed too in the subsequent commit

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx (hunk 2 above)
Comment (Daphne210) [reply src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx]: > lets stick o3 implementation style of modals and dialogs

This too

File: src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx (hunk 1 above)
Comment (Daphne210) [reply src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx]: > do we really need this tag

This has been addressed in the subsequent commit

File: src/stock-items/add-stock-item/packaging-units/packaging-units-delete-modal.component.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            `Error Deleting a stock item packing unit`
+          ),
+          kind: "error",
+          critical: true,
+          description: error?.message,
+        });
+      }
+    );
+  };
+  console.info(row.packagingUomName);
```
Comment (jabahum) [inline src/stock-items/add-stock-item/packaging-units/packaging-units-delete-modal.component.tsx:59]: can we remove this console

Comment (Daphne210) [reply src/stock-items/add-stock-item/packaging-units/packaging-units-delete-modal.component.tsx]: > can we remove this console

This has been removed. Thanks @jabahum

**Review by jabahum — APPROVED** [review]
LGTM

**Pull request merged**